packages/aws-cdk-lib/aws-apigateway/README.md

```diff
@@ -267,6 +267,10 @@ The following example uses sets up two Resources '/pets' and '/books' in separat
 
 [Resources grouped into nested stacks](test/integ.restapi-import.lit.ts)
 
+> **Warning:** With the above code, no Deployment is created even if there are changes to the resources, and the latest resources are not reflected.
+To create a Deployment, you need to use [`Deployment.addToLogicalId()`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.Deployment.html#addwbrtowbrlogicalwbriddata).
+Currently, it is possible to work around this issue with a workaround implementation. Please refer to the [related issue](https://github.com/aws/aws-cdk/issues/13526).
```

Comment:
I told you to write specific workarounds in the threads above, because I thought it's not clear to users what to do just to post the link to the issue.

If we write the workarounds, do we need to post the issue link here? Is there more information packed into this issue than that? (It's already closed, so just posting the link may cause users to close the page. If you want to include it, it might be a good idea to tell them what they should look for.)

Reply:
(Is this a bug in the first place...? If it is a bug that needs to be resolved, perhaps the best course of action is to open a new issue and work on it. If it's not a bug, then I'm not sure, but maybe there's no need to post a link to the issue here...)

Reply:
@go-to-k 

Thank you very much for the detailed review. It is challenging to determine whether this is a bug, as similar issues have been raised multiple times in the past. In those cases, workarounds have been shared, and the issues have either been closed or remain unresolved due to the time required to address them.

The reason why the deployment is not created is that the object obtained by `RestApi.fromRestApiAttributes()` does not have dynamic attributes such as `latestDaployment`.

Reference Issues:
- https://github.com/aws/aws-cdk/issues/13526#issuecomment-1011216177
- https://github.com/aws/aws-cdk/issues/12417#issuecomment-1551338526

The concern is that developers considering stack splitting may find it difficult to discover this problem.

The current issue with the documentation is that it shares the implementation of stack splitting where automatic deployment is not performed. Would it be appropriate to add the implementation of `Deployment.addToLogicalId` to the sample code? 
However, even with this implementation, the issue of creating the deployment at the appropriate timing remains.

```
class DeployStack extends NestedStack {
  constructor(scope: Construct, props: DeployStackProps) {
    super(scope, 'integ-restapi-import-DeployStack', props);

    const deployment = new Deployment(this, 'Deployment', {
      api: RestApi.fromRestApiId(this, 'RestApi', props.restApiId),
    });
    if (props.methods) {
      for (const method of props.methods) {
        deployment.node.addDependency(method);
      }
    }
    const logicalId = // salted id
    deployment.addToLogicalId(logicalId)
    new Stage(this, 'Stage', { deployment });
  }
}
```

Alternatively, if this is a long-standing issue, should we wait for a resolution, even though the timeline is uncertain?

Reply:
Thanks for clearing.

> Would it be appropriate to add the implementation of Deployment.addToLogicalId to the sample code?
> However, even with this implementation, the issue of creating the deployment at the appropriate timing remains.

I'm not sure, if the implementation is incomplete and remains the another issue, it may not be appropriate to recommend it... It would also be helpful to just state that deployment must be done manually when resources are changed. I think it's important to write only what we know completely.

> Alternatively, if this is a long-standing issue, should we wait for a resolution, even though the timeline is uncertain?

I don't see the need to wait all, as there is no guarantee that the solution will be found, and in the meantime, there will be new users who will be in trouble.